feat: API integration for login and registration page in admin app

sidarth-23 opened the pull request:
<!--
  Thanks for submitting a pull request!
  We appreciate you spending the time to work on these changes. Please provide enough information so that others can review your pull request. 
-->

## Summary
- Add form submission handling for the registration form to send a POST request to the REGISTER_URL with user data, set cookies on success, and handle errors from the server response
- Add form submission handling for the login form to send a POST request to the LOGIN_URL with user data, set cookies on success
- fix: update BASE_URL to default to `http://127.0.0.1:8000/api/` if not provided. The env is not reading properly. Hence the hard coded value for the time being for dev setup
- feat: add types for registration success and error responses, login success and error responses
- fix: layout shift in reset password page

<!--
 Explain the **motivation** of this Pull Request
-->
### Checklist
- [x] Tested (Must)
- [ ] Test Case added
- [x] Build Successful (Must)
- [x] Sufficient Code comments added (Must)
- [ ] Attached Screenshots / Videos <!-- if the PR contains UI changed, fixes, improvements -->
- [x] All Relevant Documents added


## Depends on
#77 
<!--- Does this PR depend on another PR that should be merged first or at the same time -->

yaju09 commented on apps/admin/app/(auth)/login/_components/form.tsx:
```diff
@@ -24,8 +33,24 @@ export function LoginForm() {
+      .catch((error: AxiosError<TLoginErrorResponse>) => {
+        console.log("error", error);
+        if (error.response) {
+          console.log("error.response", error.response);
```
is the console required?

yaju09 commented on apps/admin/app/(auth)/login/_components/form.tsx:
```diff
@@ -24,8 +33,24 @@ export function LoginForm() {
+        router.push("/dashboard");
+      })
+      .catch((error: AxiosError<TLoginErrorResponse>) => {
+        console.log("error", error);
```
is the console required?

yaju09 commented on apps/admin/app/(auth)/register/_components/form.tsx:
```diff
@@ -21,8 +28,44 @@ export function RegisterationForm() {
+      })
+      .catch((error: AxiosError<TRegistrationErrorResponse>) => {
+        if (error.response) {
+          error.response.data.email &&
```
```suggestion
         const error =  error.response.data
         error.email && form.setError("email", {
              type: "server",
              message: error.email[0],
            });
         ```
         is it possible to reduce the size of this function

sidarth-23 replied:
For the time being until we come up with a consistent naming scheme for frontend and backend, it is not possible to reduce the function size

sidarth-23 committed:
feat: add form submission handling for registration and login forms

- Add form submission handling for the registration form to send a POST request
  to the REGISTER_URL with user data, set cookies on success, and handle errors
  from the server response
- Add form submission handling for the login form to send a POST request to the
  LOGIN_URL with user data, set cookies on success
- fix: update BASE_URL to default to 'http://127.0.0.1:8000/api/' if not provided. The env is not reading properly. Hence the hard coded value for the time being for dev setup
- feat: add types for registration success and error responses, login success and
error responses
- fix: layout shift in reset password page
fix: update type definition for TLoginErrorResponse to expect an array of errors instead of a single string. Also, add REGISTER_URL constant to the base URL for registration functionality.
fix: update error handling in login form submission to correctly display error messages

sidarth-23 committed:
Merge branch 'master' into feat/api-integration-for-login-and-registration

sidarth-23 committed:
Merge branch 'master' into feat/api-integration-for-login-and-registration
